crates/exex/src/hints.rs

```diff
@@ -0,0 +1,137 @@
+use crate::{db::Database, exex::DATABASE_PATH};
+use cairo_vm::{
+    hint_processor::{
+        builtin_hint_processor::builtin_hint_processor_definition::{
+            BuiltinHintProcessor, HintFunc,
+        },
+        hint_processor_definition::HintReference,
+    },
+    serde::deserialize_program::ApTracking,
+    types::exec_scope::ExecutionScopes,
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+    Felt252,
+};
+use rusqlite::Connection;
+use std::{collections::HashMap, fmt, rc::Rc};
+
+/// A wrapper around [`BuiltinHintProcessor`] to manage hint registration.
+pub struct KakarotBuiltinHintProcessor {
+    /// The underlying [`BuiltinHintProcessor`].
+    processor: BuiltinHintProcessor,
+}
```

Comment:
I don't know why they call this "Builtin"HintProcess and not just `HintProcessor`. It looks like it's tied to builtins, I don't understand this

Reply:
It was just to "act" as a wrapper around the `BuiltinHintProcessor` which is quite common in Rust. But I agree on the naming, just changed